pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -0,0 +1,161 @@
+)
+
+const (
+	clusterResourceOverrideAnnotation = "quota.openshift.io/cluster-resource-override-enabled"
+	cpuBaseScaleFactor                = 1000.0 / (1024.0 * 1024.0 * 1024.0) // 1000 milliCores per 1GiB
+)
+
+func init() {
+	admission.RegisterPlugin("ClusterResourceOverride", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newClusterResourceOverride(client, config)
+	})
+}
+
+type clusterResourceOverridePlugin struct {
+	*admission.Handler
+	Config       *api.ClusterResourceOverrideConfig
+	ProjectCache *cache.ProjectCache
+	LimitRanger  admission.Interface
+}
+
+var _ = oadmission.WantsProjectCache(&clusterResourceOverridePlugin{})
+var _ = oadmission.Validator(&clusterResourceOverridePlugin{})
+
+// newClusterResourceOverride returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newClusterResourceOverride(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+	glog.V(5).Infof("ClusterResourceOverride admission controller is loaded")
+	parsed, err := ReadConfig(config)
+	glog.V(5).Infof("ClusterResourceOverride admission controller got config: %v\nerror: (%T) %[2]v", parsed, err)
+	return &clusterResourceOverridePlugin{
+		Handler:     admission.NewHandler(admission.Create),
+		Config:      parsed,
+		LimitRanger: limitranger.NewLimitRanger(client, wrapLimit),
+	}, err
+}
+
+func wrapLimit(limitRange *kapi.LimitRange, resourceName string, obj runtime.Object) error {
+	limitranger.Limit(limitRange, resourceName, obj)
+	// always return success so that all defaults will be applied.
+	// validation will occur after the overrides.
+	return nil
+}
+
+func (a *clusterResourceOverridePlugin) SetProjectCache(projectCache *cache.ProjectCache) {
+	a.ProjectCache = projectCache
+}
+
+func ReadConfig(configFile io.Reader) (*api.ClusterResourceOverrideConfig, error) {
+	if configFile == nil || reflect.ValueOf(configFile).IsNil() /* pointer to nil */ {
+		glog.V(5).Infof("ClusterResourceOverride has no config to read.")
+		return nil, nil
+	}
+	glog.V(5).Infof("ClusterResourceOverride about to read config:\n%v", configFile)
+	buffer := new(bytes.Buffer)
+	if _, err := buffer.ReadFrom(configFile); err != nil {
+		return nil, err
+	}
+	config := &api.ClusterResourceOverrideConfig{}
+	err := yaml.Unmarshal(buffer.Bytes(), config)
+	glog.V(5).Infof("ClusterResourceOverride config:\n%v\nerror: %v", config, err)
+	return config, err
+}
+
+func (a *clusterResourceOverridePlugin) Validate() error {
+	if err := validate.Validate(a.Config); err != nil {
+		return err
+	}
+	if a.ProjectCache == nil {
+		return fmt.Errorf("ClusterResourceOverride did not get a project cache")
+	}
+	return nil
+}
+
+// TODO this will need to update when we have pod requests/limits
+func (a *clusterResourceOverridePlugin) Admit(attr admission.Attributes) error {
+	glog.V(8).Infof("ClusterResourceOverride admission controller is invoked")
```

Comment:
Try avoid going below 6.  The client debugging gets really noisy past 6 and we made it that high to avoid it spamming detailed logs from other components.